nifi-docs/src/main/asciidoc/user-guide.adoc

```diff
@@ -729,23 +901,30 @@ NiFi properties have resolution precedence of which you should be aware when cre
 * Processor-specific attributes
 * FlowFile properties 
 * FlowFile attributes 
-* From variable registry: 
+* From Variable Registry: 
 ** User defined properties (custom properties)
 ** System properties 
 ** Operating System environment variables
 
 When you are creating custom properties, ensure that each custom property contains a distinct property value,
 so that it is not overridden by existing environment properties, system properties, or FlowFile attributes.
 
-There are two ways to use and manage custom properties:
+There are two ways to use and manage custom properties with Expression Language:
+
+- *Variables*: Variables are created and configured within the NiFi UI. They can be used in any field that supports Expression Language. Variables cannot be used for sensitive properties. NiFi automatically picks up new or modified variables. Variables are defined at the Process Group level, as a result, the access policies for viewing and changing variables are derived from the access policies of the Process Group. See <<Variables>> for more information.
+
+WARNING: Variables are still supported for compatibility purposes but do not have the same power as Parameters such as support for sensitive properties and more granular control over who can create, modify or use them. Variables will be removed in a future release. As a result, it is highly recommended to switch to Parameters.
```

Comment:
Maybe this note should also be under the `Variables` heading (lines 920-925)? The link from the top of the doc takes you there, you might miss this important piece of info.

Reply:
Good call. Added this warning under the "Variables" heading and the "Referencing Custom Properties via nifi.properties" heading.